### DISCUSSION TITLE
About compile

### DISCUSSION THREAD

**Discussion by cosmoquester** [Stellarium/stellarium#181] [Q&A]
I saw wiki of this and try compile several method that represented in wiki.
But I cannot success compile this.
I think information about compile on this wiki is not recent.
well, Is there recent information about compilation?

**Comment by alex-w** [comment]
The information is not recent but it valid. Please inform us about your operating system?

**Comment by cosmoquester** [comment]
My OS is Windows 10 64bit.

**Comment by alex-w** [comment]
OK. Could you describe step when you got compiling issue?

**Comment by gzotti** [comment]
Follow the guidelines https://github.com/Stellarium/stellarium/wiki/Compilation-on-Windows but use MSVC2015 [free edition is OK] or later. Install the appropriate version of Qt, current may be 5.10. And build inside QtCreator. 

If this does not work roughly as described, describe what you did to fail.

**Comment by cosmoquester** [comment]
Thank you, vs2013 puts some error about not supported code 
but I success compile using VS2017.
I think wiki windows compile page need to be updated.
May I modify that windows compile page? maybe about Visual studio compile contents

**Comment by gzotti** [comment]
I would say, yes please. Esp. the part for experienced users of MSVC (who then use MSVC IDE and not QtCreator for development) needs verification/correction. I only installed MSVC2015 to have the compiler.  
I hope MSVS2015 and 2017 work then similarly. MSVC2013 can be dropped (has been dropped for Qt5.11).

Also, description of MinGW build of dynamic plugins should be re-verified.

**Comment by cosmoquester** [comment]
OK, then I will test and update some information for windows compile wiki page.

I just modify one sector, but I think It is better that I modify at one go.

**Comment by cosmoquester** [comment]
I tested visual studio build and making installer process.
so, I updated Compilation on Windows wiki page.
but I can't 'Building with QtCreator'. 
I only success with visual studio. So This part is the same.
and 'Doing a test run' part is not work for me.
I excute compiled file. But that puts SpoutLibrary.dll miss error.
so I put that dll, and excute. But It is not work. 
No appear screen.  So, this part is also the same.

I success testing making installer, installing other computer and using that stellarium.

**Comment by gzotti** [comment]
OK, thanks. So at least building with MSVC works "as documented" now. I added some lines to the QtCreator part. However, one issue may be code style (indentation, tabs/space etc.). Is there a way to specify that in MSVC IDE?

"Doing a test run" should work for all ways: create a link in the source dir which points to the .exe file, but executes in the sources dir. 

@alex-w I have seen there is a Qt installer now. Do you have any plans to use that instead of Inno? (would remove one external dependency.)

**Comment by cosmoquester** [comment]
on visual studio, Tools - Options -> Text Editor -> C/C++ 
You can set some code style.

**Comment by gzotti** [comment]
Ah, thanks. Many options, indeed. Is there a way to load all at once? (And then someone would have to create such a file to avoid indentation mess on the GIT...)

**Comment by cosmoquester** [comment]
Umm....In Edit - Advanced, you can align selected specific source code file.
But I don't know about loading all at once...

**Comment by gzotti** [comment]
OK, no problem. Thanks for the updates so far. If you have further hints&corrections, please feel free to edit. If not, please close the issue.

**Comment by alex-w** [comment]
@gzotti no plans to switch to Qt Installer Framework. Are you want to rewrite rules for installer?

**Comment by gzotti** [comment]
@alex-w not necessarily. Just if you had such plans, updating Inno instructions in the wiki would have been pointless.
